### PR TITLE
feature(framework): keras add save_format option

### PR DESCRIPTION
**Pull request by larme**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/frameworks/keras.py [hunk 1]
```diff
@@ -132,29 +138,41 @@ def load(
...
-            with Path(model.path_of(_MODEL_JSON_FNAME)).open("r") as jsonf:
+        if store_as_json:
+            assert Path(bentoml_model.path_of(_MODEL_JSON_FNAME)).is_file()
+            with Path(bentoml_model.path_of(_MODEL_JSON_FNAME)).open("r") as jsonf:
                 model_json = jsonf.read()
             model = keras.models.model_from_json(
                 model_json, custom_objects=default_custom_objects
             )
+            weight_fname = _MODEL_WEIGHT_FNAME_MAPPING[save_format]
+            model.load_weights(bentoml_model.path_of(weight_fname))
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply bentoml/_internal/frameworks/keras.py:171]: We shouldn't load weight when `store_as_json=True`.

File: bentoml/_internal/frameworks/keras.py [hunk 2]
```diff
@@ -184,6 +203,8 @@ def save(
             Instance of the Keras model to be saved to BentoML modelstore.
         store_as_json (`bool`, `optional`, default to :code:`False`):
             Whether to store Keras model as JSON and weights.
+        save_format (`str`, `optional`, default to :code:`tf`):
+            Whether to store Keras model or weight in tf format or old h5 format.
```
[the remark this answers is not in the document]
Comment (larme) [reply bentoml/_internal/frameworks/keras.py:207]: yep I think it's what tf document encourage us to do

File: bentoml/_internal/frameworks/keras.py [hunk 3]
```diff
@@ -132,29 +138,41 @@ def load(
...
+            with Path(bentoml_model.path_of(_MODEL_JSON_FNAME)).open("r") as jsonf:
                 model_json = jsonf.read()
             model = keras.models.model_from_json(
                 model_json, custom_objects=default_custom_objects
             )
+            weight_fname = _MODEL_WEIGHT_FNAME_MAPPING[save_format]
+            model.load_weights(bentoml_model.path_of(weight_fname))
         else:
+            model_fname = _SAVED_MODEL_FNAME_MAPPING[save_format]
             model = keras.models.load_model(
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/keras.py:174]: does `load_model` accept a `SavedModel` format?

Comment (aarnphm) [reply bentoml/_internal/frameworks/keras.py:174]: We are also planning to have a `import_tf_savedmodel` for tensorflow as well as `import_from_pt` for pytorch

**Review by aarnphm — COMMENTED** [review]
https://www.tensorflow.org/guide/keras/save_and_serialize#whole-model_saving_loading. Should we separate into `save_as_h5` and `save_as_savedmodel`?

**Review by aarnphm — APPROVED** [review]
👍🏼

**Pull request merged**